src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListener.cs

```diff
@@ -64,19 +71,23 @@ public void OnNext(KeyValuePair<string, object> kv)
 
 					if (ProcessingRequests.TryAdd(request, span))
 					{
+						if(!request.Headers.Contains(TraceParent.TraceParentHeaderName))
```

Comment:
The expected behaviour is to have a trace with multiple transactions. It'll be tested in https://github.com/elastic/apm-agent-dotnet/issues/42 with other agents. In this PR it's only covered across 2 .NET processes in `Elastic.Apm.AspNetCore.Tests.DistributedTracingAspNetCoreTests.DistributedTraceAcross2Service`

Reply:
I'm not sure we are on the same page - IIUC the `if` above is to test if `traceparent` header is already present in _outgoing_ HTTP request (maybe application has another APM tool or for some other reason). The question is what is the expected agent's behavior in this case?

Reply:
Och, now I see. Yeah, we can follow up with them, but I'm not sure we have a solid plan for this, since the header name is non standard, so in the current phase this should not be a problem. On the other hand I saw this happening during testing, maybe we got multiple callbacks for a single outgoing requests. I'll follow up on this one.

Reply:
Ok, now I know what happened. In the integration test accidentally I subscribed to the diagnostic source messages twice. So I got two `System.Net.Http.HttpRequestOut.Start` events per HTTP call. In this case without this if block the wrong one won.

This is totally my fault and should not happen, but that is already a scenario, where this if block helps. I'd say if there is already 1 header we should not add it again. I'd feel better by keeping that if block there.

Reply:
Essentially my concern is that when we switch to the standard header name(s) a customer might have another APM tool that also injects `traceparent` header and the customer might want our agent to overwrite the header's value. Maybe because the customer thinks our APM solution better but the customer cannot disable the other APM tool for some reason. There will be no way to make our agent behave this way. Do you think it's an interesting use case to consider?

Reply:
Yeah, we had a short discussion about https://github.com/elastic/apm/issues/71 and this came up. I opened https://github.com/elastic/apm-agent-dotnet/issues/177 for .NET.

I think as long as the standard does not say anything about multiple tools in the same process we should not invent something smart. Once someone does that there will be all sorts of other problems. I think this is ok for the prefixed flag, since only we use that, and in https://github.com/elastic/apm-agent-dotnet/issues/177 we can address this if needed. I think https://github.com/elastic/apm/issues/71 would be the right place to figure out a strategy around across all agents.